### PR TITLE
feat: implement deepContains function

### PR DESCRIPTION
**Pull request by dbouwman**
1. Description:

Addresses [4868](https://devtopia.esri.com/dc/hub/issues/4868), which is part of [4802](https://devtopia.esri.com/dc/hub/issues/4802).

Note: `deepContains` is specifically _not_ exported. It will be surfaced through `HubSite.contains(...)`, which will be in a subsequent PR

1. Instructions for testing:

- run test

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

### CODE REVIEW

**Review by tomwayson — APPROVED** [review]
LGTM other than the duplicate line

File: packages/common/src/core/_internal/deepContains.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ * - Check if `00b` is contained in `00a`'s catalog
+ * If all three are true, then `00d` is considered contained in `00a`'s catalog.
+ *
+ * @param identifier
+ * @param options
+ * @returns
+ */
+export async function deepContains(
+  identifier: string,
+  hiearchy: IDeepCatalogInfo[],
```
Comment (drspacemanphd) [inline packages/common/src/core/_internal/deepContains.ts:30]: minor but I believe this should be _hierarchy_ in all references.

**Review by drspacemanphd — APPROVED** [review]

File: packages/common/src/core/_internal/deepContains.ts (hunk 1 above)
Comment (dbouwman) [reply packages/common/src/core/_internal/deepContains.ts:30]: As if `initiative` was not bad enough, I'm adding `hierarchy` 🤦🏼‍♂️  Open to other terms... but thanks for catching this

File: packages/common/src/index.ts [hunk 2]
```diff
@@ -14,26 +14,25 @@ export * from "./downloads";
...
+export * from "./pages";
 export * from "./projects";
-export * from "./initiatives";
 export * from "./request";
 export * from "./resources";
 export * from "./search";
 export * from "./sites";
 export * from "./surveys";
 export * from "./types";
+export * from "./types";
```
[the remark this answers is not in the document]
Comment (dbouwman) [reply packages/common/src/index.ts:30]: ha! What's funny is that all I did in that file was sort the lines... so the dupe was already there, but not visible b/c it was not sorted. Will remove.

**Pull request merged**